File: gems/decomplex/examples/lua/locality-drag.lua
function run(user, cart, logger)
  local receipt_id = user.id

  local total = cart.total
  if total > 100 then
    if cart.discountable() then
      local discount = 10
    end
  end
  if cart.taxable() then
    if cart.region then
      local tax = total * 2
    end
  end
  if logger.enabled() then
    if logger.debug() then
      logger.info(total)
    end
  end
  if cart.valid() then
    if cart.ready() then
      local status = 1
    end
  end

  emit(receipt_id)
end

File: gems/decomplex/examples/c/locality-drag.c
void run(User user, Cart cart, Logger logger) {
  int receipt_id = user.id;

  int total = cart.total;
  if (total > 100) {
    if (cart.discountable()) {
      int discount = 10;
    }
  }
  if (cart.taxable()) {
    if (cart.region) {
      int tax = total * 2;
    }
  }
  if (logger.enabled()) {
    if (logger.debug()) {
      logger.info(total);
    }
  }
  if (cart.valid()) {
    if (cart.ready()) {
      int status = 1;
    }
  }

  emit(receipt_id);
}
